src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -60,9 +60,11 @@ internal class ConnectionManager : IConnectionManager {
                 .Add(_statusBarViewModel)
                 .Add(_hostLoadIndicatorViewModel)
                 .Add(() => _sessionProvider.BrokerStateChanged -= BrokerStateChanged)
+                .Add(() => _sessionProvider.SessionStateChanged -= SessionStateChanged)
```

Comment:
It would've make sense if there was more than one session that we are interested in this case. But yellow status is specific to REPL. So I'd prefer to explicitly handle `_interactiveWorkflow.RSession.Connected` and `_interactiveWorkflow.RSession.Disconnected` events without `sessionProvider` being a proxy.